packages/@react-aria/table/src/useTable.ts

```diff
@@ -109,7 +109,52 @@ export function useTable<T>(props: TableProps<T>, state: TableState<T>, ref: Ref
     return sortDirection && column ? formatMessage(`${sortDirection}Sort`, {columnName}) : undefined;
   }, [sortDirection, column, state.collection.columns]);
 
-  let descriptionProps = useDescription(sortDescription);
+
+  // do we have something that might be better suited for this? issue is that useInteractionModality
+  // doesn't return quite enough information
+  let [interactionType, setInteractionType] = useState('');
+  useEffect(() => {
```

Comment:
This logic should probably be shared with other components. It will be needed in `useGrid` at the very least (for ListView etc.), but really will be needed for any component that has highlight selection. For other component like ListBox, we may want the description on the actual options rather than the list itself (if this works in screen readers).  Maybe a hook in the selection package for the description computation, plus built into `useGrid`? We should test highlight selection in listbox as well (not Spectrum).